src/Illuminate/Http/Concerns/InteractsWithInput.php

```diff
@@ -94,15 +94,17 @@ public function has($key)
     /**
      * Determine if the request contains any of the given inputs.
      *
-     * @param  dynamic  $key
+     * @param  string|array  $key
      * @return bool
      */
-    public function hasAny(...$keys)
+    public function hasAny($key)
```

Comment:
```php
public function hasAny(...$keys)
{
    if (isset($keys[0]) && is_array($keys[0])) {
        $keys = $keys[0];
    }
```

Reply:
I agree that the method signature could be kept and avoid that specific breaking change; but some further digging shows that we already support arrays to Request::hasAny(), so a change like this would introduce other problems.